src/Serialization/Rings.jl

```diff
@@ -0,0 +1,53 @@
+################################################################################
+# fmpq polynomial ring
+function save_internal(s::SerializerState, R::FmpqMPolyRing)
+    base_ring = R.base_ring
+    return Dict(
+        :symbols => save_type_dispatch(s, [String(x) for x in R.S]),
```

Comment:
Perhaps we can add a (de)serializer for symbols then this could simply become
```suggestion
        :symbols => save_type_dispatch(s, R.S),
```